### PR TITLE
`incremental_save` fixes

### PR DESCRIPTION
**Pull request by carmocca**
See comments

### CODE REVIEW

File: scripts/convert_hf_checkpoint.py [hunk 1]
```diff
@@ -83,66 +78,57 @@ def permute(w):
...
-                    sd_key = weight_map[name]
-                    sd[sd_key] = param._load_tensor().to(dtype)
-                if not skip:
-                    sd[sd_key] = output_file.store_early(sd[sd_key])
-
-            gc.collect()
-
-        output_file.save(sd)
-
-    assert len(unprocessed_weights) == 0, f"unexpected partial weights {unprocessed_weights.key()}"
```
Comment (carmocca) [inline scripts/convert_hf_checkpoint.py:145]: replaced `.key()` which doesn't exist

File: scripts/convert_hf_checkpoint.py [hunk 2]
```diff
@@ -83,66 +78,57 @@ def permute(w):
...
+                                .to(dtype)
+                            )
+                            sd[sd_key] = w
+                            del unprocessed_weights[sd_key]
+                            skip = False
+                        else:
+                            sd[sd_key] = param._load_tensor().to(dtype)
                     else:
+                        sd_key = weight_map[name]
                         sd[sd_key] = param._load_tensor().to(dtype)
```
Comment (carmocca) [inline scripts/convert_hf_checkpoint.py:125]: This else in the previous code is now removed, it should fall under the L121 else in the new code

File: scripts/convert_hf_checkpoint.py [hunk 3]
```diff
@@ -50,12 +48,9 @@ def convert_hf_checkpoint(
...
 
-    # Load the json file containing weight mapping
-    pytorch_bin_map_json_path = checkpoint_dir / "pytorch_model.bin.index.json"
-    with open(pytorch_bin_map_json_path) as json_map:
-        bin_index = json.load(json_map)
-
-    bin_files = set(el for el in bin_index["weight_map"].values())
+    bin_files = list(checkpoint_dir.glob("*.bin"))
+    if not bin_files:
+        raise ValueError(f"Expected {str(checkpoint_dir)!r} to contain .bin files")
```
Comment (carmocca) [inline scripts/convert_hf_checkpoint.py:53]: Simplifying, we do this in lit-parrot

File: scripts/convert_hf_checkpoint.py [hunk 4]
```diff
@@ -83,66 +78,57 @@ def permute(w):
 
     print(f"Saving to disk at {output_dir}")
     unprocessed_weights = collections.defaultdict(dict)
-    with contextlib.ExitStack() as stack:
```
Comment (carmocca) [inline scripts/convert_hf_checkpoint.py:86]: No ExitStack

File: lit_llama/utils.py [hunk 5]
```diff
@@ -327,13 +327,11 @@ def __init__(self, obj, saver, protocol_version=5):
...
             # PT upstream wants to deprecate this eventually...
             storage = obj._untyped_storage
-            storage_dtype = obj.dtype
             storage_type_str = obj._pickle_storage_type()
             storage_type = getattr(torch, storage_type_str)
             storage_numel = obj._size()
         else:
             storage = obj
-            storage_dtype = torch.uint8
             storage_type = normalize_storage_type(type(obj))
```
Comment (carmocca) [inline lit_llama/utils.py:335]: The import for this was missing, so clearly this code path is not used.

File: scripts/convert_hf_checkpoint.py (hunk 1 above)
Comment (t-vi) [reply scripts/convert_hf_checkpoint.py:145]: oops. Thank you.

File: scripts/convert_hf_checkpoint.py (hunk 3 above)
Comment (t-vi) [reply scripts/convert_hf_checkpoint.py:54]: Seems OK per se, but I wonder if there might be instances where users have other bin files for no good reason (e.g. an older version of their model had a split into smaller files).

File: lit_llama/utils.py (hunk 5 above)
Comment (t-vi) [reply lit_llama/utils.py:335]: Yes, and removing the unused `storage_dtype` is great but I would expect a future update of Pytorch to hit the second part. (but `normalize_storage_type` needs to be qualified with `torch.serialization`.... .ehem.)

**Review by t-vi — APPROVED** [review]
Some comments but looks great. Thank you! (Not that my review counts. :) )

**Review by awaelchli — APPROVED** [review]
Looks good, if good with @t-vi

File: scripts/convert_hf_checkpoint.py (hunk 3 above)
Comment (carmocca) [reply scripts/convert_hf_checkpoint.py:54]: So far I haven't seen any hf-hub checkpoint that has that.

Comment (carmocca) [reply scripts/convert_hf_checkpoint.py:54]: That's a good point. I'll revert it

**Pull request merged**